### PR TITLE
Bump fsspec version

### PR DESCRIPTION
**Pull request by skshetry**
Testing new review assignment

* [ ] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [ ] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

**Comment by skshetry** [comment]
We need to wait for adlfs and gcsfs to make a release.

**Review by karajan1001 — APPROVED** [review]
Conda is still on `2021.6.0`?

**Comment by efiop** [comment]
@karajan1001 Yeah, it will get updated automatically a bit later today.

**Comment by efiop** [comment]
Ok, introduced a temporary hack for now.

File: dvc/fs/local.py [hunk 1]
```diff
@@ -32,7 +32,9 @@ def open(path_info, mode="r", encoding=None, **kwargs):
 
     def exists(self, path_info) -> bool:
         assert isinstance(path_info, str) or path_info.scheme == "local"
-        return self.fs.exists(path_info)
+        # NOTE: os.fspath could be removed after fsspec 2021.06.1
+        # https://github.com/intake/filesystem_spec/issues/666
+        return self.fs.exists(os.fspath(path_info))
```
Comment (skshetry) [inline dvc/fs/local.py:37]: Do we need to use `fsspec` here? I think this is the only place where we use it.

Comment (efiop) [reply dvc/fs/local.py:37]: @skshetry The plan is to slowly start using fsspec's localfs instead of ours. So it is not necessary right now, sure.

**Pull request merged**